compiler/rustc_lint/src/types.rs

```diff
@@ -727,7 +728,48 @@ declare_lint! {
     "proper use of libc types in foreign item definitions"
 }
 
-declare_lint_pass!(ImproperCTypesDefinitions => [IMPROPER_CTYPES_DEFINITIONS]);
+declare_lint! {
+    /// In it's platform C ABI, the AIX target follows the power alignment rule
+    /// for structs. This rule applies the natural alignment of the first struct
+    /// field if it is a floating-point type that is greater than 4-bytes, or
+    /// is an aggregate whose recursively "first" field is a floating-point
+    /// type greater than 4-bytes.
+    /// The alignment for subsequent fields of the associated structs use an
+    /// alignment value where the floating-point type is aligned on a 4-byte
+    /// boundary.
+    /// In this case, "natural alignment" (detailed in
+    /// https://www.ibm.com/docs/en/xl-c-and-cpp-aix/16.1?topic=data-using-alignment-modes#alignment)
```

Comment:
This link should probably come first, actually.

Reply:
Sorry for the silly question, but just wanted to check if I understand the suggestion. Do you mean that the link should be in the paragraph above instead and we should remove this section (since your suggestion above relates to having to avoid describing natural alignment)?

Please do correct me if I misunderstood. :-)

Reply:
Correct, it should probably be on the first line.

Reply:
Sounds good. Thanks for clarifying! I tried to make it closer to the top/first line, so hopefully you are okay with its placement.